msdk-io-mgf/src/main/java/io/github/msdk/io/mgf/MgfFileImportMethod.java

```diff
@@ -0,0 +1,168 @@
+package io.github.msdk.io.mgf;
+
+import io.github.msdk.MSDKException;
+import io.github.msdk.MSDKMethod;
+import io.github.msdk.datamodel.MsSpectrumType;
+import io.github.msdk.util.ArrayUtil;
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.Collection;
+import java.util.Hashtable;
+import java.util.LinkedList;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/*
+ * (C) Copyright 2015-2018 by MSDK Development Team
+ *
+ * This software is dual-licensed under either
+ *
+ * (a) the terms of the GNU Lesser General Public License version 2.1 as published by the Free
+ * Software Foundation
+ *
+ * or (per the licensee's choosing)
+ *
+ * (b) the terms of the Eclipse Public License v1.0 as published by the Eclipse Foundation.
+ */
+public class MgfFileImportMethod implements MSDKMethod<Collection<MgfMsSpectrum>> {
+
+  private final Logger logger = LoggerFactory.getLogger(this.getClass());
+  private Collection<MgfMsSpectrum> spectra;
+  private final @Nonnull File target;
+  private boolean canceled = false;
+  private long processedSpectra = 0;
+  private Hashtable<String, Pattern> patterns;
+
```

Comment:
I think it is a good practice to put the class constructor as the first method (move it to here)